src/pages/admin/[cid]/topics.tsx

```diff
@@ -0,0 +1,236 @@
+import { Label } from '@components/basic/Label'
+import { RootState } from '@redux/store'
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { request } from '@utils/api'
+import { LoadClassInfoParams, LoadClassInfoResults } from '@api/admin/loadClassInfo'
+import styled from '@emotion/styled'
+import { palette, typography } from '@styles/theme'
+import { TABLET_WIDTH_THRESHOLD } from 'src/constants/ui'
+import { FillBtn, TextBtnCta } from '@components/basic/button/Button'
+import { UpdateTopicInfoParams, UpdateTopicInfoResults } from '@api/admin/updateTopicInfo'
+import { UpdateTopicDialog } from '@components/Dialogs/updateTopicDialog'
+import { CheckDialog } from '@components/Dialogs/CheckDialog'
+
+export default function Page() {
+  const isAdmin = useSelector((state: RootState) => state.userInfo.isAdmin)
```

Comment:
Perhaps, a more secure way is to block users from entering the page. We can do this with server side props
https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props#redirect

Reply:
we can change this later